### ISSUE TITLE
Cannot install MongoDB, brew install MongoDB breaks at Cheetah

### ISSUE THREAD

**Issue by tonglianheng** [Homebrew/homebrew-core#29623]
brew install mongodb

fails, with the error

Last 15 lines from /Users/tonglh/Library/Logs/Homebrew/mongodb/01.python:
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/private/tmp/mongodb-20180701-89209-1lwwrgw/mongodb-src-r4.0.0/vendor
--single-version-externally-managed
--record=installed.txt

Traceback (most recent call last):
  File "<string>", line 4, in <module>
  File "setup.py", line 10, in <module>
    import SetupTools
  File "/private/tmp/mongodb--Cheetah-20180701-89209-47q2xf/Cheetah-2.4.4/SetupTools.py", line 50
    except DistutilsPlatformError, x:
                                 ^
SyntaxError: invalid syntax

---------------------

It seems that for some reason Cheetah is forcing the system to use Python 3, which it does not support.  I am pretty sure the python I am using is Python 2:

command -v python
/Users/tonglh/homebrew/bin/python
sleipnir: tonglh$ python --version
Python 2.7.15

**Comment by tonglianheng** [comment]
BTW, redirecting PATH to use /usr/bin/python does not solve the problem.

Cheetah still fails with the same python 3 error message!

**Comment by tonglianheng** [comment]
Installation worked only after I have removed all traces of Python 3 from bin paths

**Comment by mistydemeo** [comment]
Can you please read the [troubleshooting](https://docs.brew.sh/Troubleshooting) guide and provide all of the requested information?

**Comment by LanikSJ** [comment]
`100% 2018-07-07 09:33:59 |ruby-2.5.1| LanikSJ in ~
± |master ✓| → brew install mongodb --debug
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/scons.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python@2.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sphinx-doc.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gdbm.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/makedepend.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/readline.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sqlite.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python@2.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gdbm.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/readline.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sqlite.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/scons.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sphinx-doc.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/makedepend.rb
==> Downloading https://fastdl.mongodb.org/src/mongodb-src-r4.0.0.tar.gz
Already downloaded: /Users/LanikSJ/Library/Caches/Homebrew/mongodb-4.0.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/cd/b0/c2d700252fc251e91c08639ff41a8a5203b627f4e0a2ae18a6b662ab32ea/Cheetah-2.4.4.tar.gz
Already downloaded: /Users/LanikSJ/Library/Caches/Homebrew/mongodb--Cheetah-2.4.4.tar.gz
Cheetah-2.4.4/cheetah/c/cheetah.h: Can't unlink already-existing object
tar: Error exit delayed from previous errors.
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
ErrorDuringExecution: Failure while executing: tar xf /Users/LanikSJ/Library/Caches/Homebrew/mongodb--Cheetah-2.4.4.tar.gz -C /private/tmp/mongodb--Cheetah-20180707-79793-1gsnc9p
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 3
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
/usr/local/Homebrew/Library/Homebrew/utils.rb:299:in `safe_system'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:86:in `safe_system'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:197:in `stage'
/usr/local/Homebrew/Library/Homebrew/resource.rb:115:in `block in unpack'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:12:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/resource.rb:114:in `unpack'
/usr/local/Homebrew/Library/Homebrew/resource.rb:92:in `stage'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb:47:in `block in install'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb:46:in `each'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb:46:in `install'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:22:in `block in install'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:92:in `debrew'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:22:in `install'
/usr/local/Homebrew/Library/Homebrew/build.rb:142:in `block (2 levels) in install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1116:in `block in brew'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1958:in `block (2 levels) in stage'
/usr/local/Homebrew/Library/Homebrew/utils.rb:557:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1957:in `block in stage'
/usr/local/Homebrew/Library/Homebrew/resource.rb:119:in `block in unpack'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:12:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/resource.rb:114:in `unpack'
/usr/local/Homebrew/Library/Homebrew/resource.rb:92:in `stage'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/lib/ruby/2.3.0/forwardable.rb:189:in `stage'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1935:in `stage'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1111:in `brew'
/usr/local/Homebrew/Library/Homebrew/build.rb:113:in `block in install'
/usr/local/Homebrew/Library/Homebrew/utils.rb:557:in `with_env'
/usr/local/Homebrew/Library/Homebrew/build.rb:110:in `install'
/usr/local/Homebrew/Library/Homebrew/build.rb:191:in `<main>'
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 1
==> Kept temporary files
Temporary files retained at /private/tmp/mongodb-20180707-79793-hrhebg
Error: Failure while executing: tar xf /Users/LanikSJ/Library/Caches/Homebrew/mongodb--Cheetah-2.4.4.tar.gz -C /private/tmp/mongodb--Cheetah-20180707-79793-1gsnc9p
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
/usr/local/Homebrew/Library/Homebrew/utils.rb:299:in `safe_system'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:86:in `safe_system'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:197:in `stage'
/usr/local/Homebrew/Library/Homebrew/resource.rb:115:in `block in unpack'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:12:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/resource.rb:114:in `unpack'
/usr/local/Homebrew/Library/Homebrew/resource.rb:92:in `stage'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb:47:in `block in install'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb:46:in `each'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mongodb.rb:46:in `install'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:22:in `block in install'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:92:in `debrew'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:22:in `install'
/usr/local/Homebrew/Library/Homebrew/build.rb:142:in `block (2 levels) in install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1116:in `block in brew'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1958:in `block (2 levels) in stage'
/usr/local/Homebrew/Library/Homebrew/utils.rb:557:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1957:in `block in stage'
/usr/local/Homebrew/Library/Homebrew/resource.rb:119:in `block in unpack'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:73:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:12:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/resource.rb:114:in `unpack'
/usr/local/Homebrew/Library/Homebrew/resource.rb:92:in `stage'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1935:in `stage'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1111:in `brew'
/usr/local/Homebrew/Library/Homebrew/build.rb:113:in `block in install'
/usr/local/Homebrew/Library/Homebrew/utils.rb:557:in `with_env'
/usr/local/Homebrew/Library/Homebrew/build.rb:110:in `install'
/usr/local/Homebrew/Library/Homebrew/build.rb:191:in `<main>'

100% 2018-07-07 09:38:13 |ruby-2.5.1| LanikSJ in ~
± |master ✓| → tar xzvvf /Users/LanikSJ/Library/Caches/Homebrew/mongodb--Cheetah-2.4.4.tar.gz -C /private/tmp/mongodb--Cheetah-20180707-79793-1gsnc9p
x Cheetah-2.4.4/
x Cheetah-2.4.4/Cheetah.egg-info/
x Cheetah-2.4.4/Cheetah.egg-info/dependency_links.txt
x Cheetah-2.4.4/Cheetah.egg-info/SOURCES.txt
x Cheetah-2.4.4/Cheetah.egg-info/requires.txt
x Cheetah-2.4.4/Cheetah.egg-info/top_level.txt
x Cheetah-2.4.4/Cheetah.egg-info/PKG-INFO
x Cheetah-2.4.4/setup.py
x Cheetah-2.4.4/TODO
x Cheetah-2.4.4/cheetah/
x Cheetah-2.4.4/cheetah/TemplateCmdLineIface.py
x Cheetah-2.4.4/cheetah/Macros/
x Cheetah-2.4.4/cheetah/Macros/__init__.py
x Cheetah-2.4.4/cheetah/Macros/I18n.py
x Cheetah-2.4.4/cheetah/Template.py
x Cheetah-2.4.4/cheetah/Tests/
x Cheetah-2.4.4/cheetah/Tests/Template.py
x Cheetah-2.4.4/cheetah/Tests/SyntaxAndOutput.py
x Cheetah-2.4.4/cheetah/Tests/Parser.py
x Cheetah-2.4.4/cheetah/Tests/Cheps.py
x Cheetah-2.4.4/cheetah/Tests/Regressions.py
x Cheetah-2.4.4/cheetah/Tests/__init__.py
x Cheetah-2.4.4/cheetah/Tests/Analyzer.py
x Cheetah-2.4.4/cheetah/Tests/Misc.py
x Cheetah-2.4.4/cheetah/Tests/Performance.py
x Cheetah-2.4.4/cheetah/Tests/Test.py
x Cheetah-2.4.4/cheetah/Tests/xmlrunner.py
x Cheetah-2.4.4/cheetah/Tests/Filters.py
x Cheetah-2.4.4/cheetah/Tests/NameMapper.py
x Cheetah-2.4.4/cheetah/Tests/Unicode.py
x Cheetah-2.4.4/cheetah/Tests/CheetahWrapper.py
x Cheetah-2.4.4/cheetah/SettingsManager.py
x Cheetah-2.4.4/cheetah/Utils/
x Cheetah-2.4.4/cheetah/Utils/WebInputMixin.py
x Cheetah-2.4.4/cheetah/Utils/statprof.py
x Cheetah-2.4.4/cheetah/Utils/Indenter.py
x Cheetah-2.4.4/cheetah/Utils/htmlEncode.py
x Cheetah-2.4.4/cheetah/Utils/__init__.py
x Cheetah-2.4.4/cheetah/Utils/Misc.py
x Cheetah-2.4.4/cheetah/Utils/htmlDecode.py
x Cheetah-2.4.4/cheetah/ImportHooks.py
x Cheetah-2.4.4/cheetah/Templates/
x Cheetah-2.4.4/cheetah/Templates/SkeletonPage.py
x Cheetah-2.4.4/cheetah/Templates/__init__.py
x Cheetah-2.4.4/cheetah/Templates/_SkeletonPage.py
x Cheetah-2.4.4/cheetah/Templates/SkeletonPage.tmpl
x Cheetah-2.4.4/cheetah/CacheRegion.py
x Cheetah-2.4.4/cheetah/Version.py
x Cheetah-2.4.4/cheetah/Parser.py
x Cheetah-2.4.4/cheetah/convertTmplPathToModuleName.py
x Cheetah-2.4.4/cheetah/Django.py
x Cheetah-2.4.4/cheetah/Tools/
x Cheetah-2.4.4/cheetah/Tools/MondoReport.py
x Cheetah-2.4.4/cheetah/Tools/__init__.py
x Cheetah-2.4.4/cheetah/Tools/SiteHierarchy.py
x Cheetah-2.4.4/cheetah/Tools/MondoReportDoc.txt
x Cheetah-2.4.4/cheetah/Tools/CGITemplate.py
x Cheetah-2.4.4/cheetah/Tools/RecursiveNull.py
x Cheetah-2.4.4/cheetah/Tools/turbocheetah/
x Cheetah-2.4.4/cheetah/Tools/turbocheetah/tests/
x Cheetah-2.4.4/cheetah/Tools/turbocheetah/tests/__init__.py
x Cheetah-2.4.4/cheetah/Tools/turbocheetah/tests/test_template.py
x Cheetah-2.4.4/cheetah/Tools/turbocheetah/__init__.py
x Cheetah-2.4.4/cheetah/Tools/turbocheetah/cheetahsupport.py
x Cheetah-2.4.4/cheetah/__init__.py
x Cheetah-2.4.4/cheetah/Servlet.py
x Cheetah-2.4.4/cheetah/SourceReader.py
x Cheetah-2.4.4/cheetah/DummyTransaction.py
x Cheetah-2.4.4/cheetah/ErrorCatchers.py
x Cheetah-2.4.4/cheetah/DirectiveAnalyzer.py
x Cheetah-2.4.4/cheetah/ImportManager.py
x Cheetah-2.4.4/cheetah/Unspecified.py
x Cheetah-2.4.4/cheetah/CacheStore.py
x Cheetah-2.4.4/cheetah/Filters.py
x Cheetah-2.4.4/cheetah/FileUtils.py
x Cheetah-2.4.4/cheetah/c/
**x Cheetah-2.4.4/cheetah/c/Cheetah.h
x Cheetah-2.4.4/cheetah/c/cheetah.h: Can't unlink already-existing object**
x Cheetah-2.4.4/cheetah/c/_namemapper.c
x Cheetah-2.4.4/cheetah/NameMapper.py
x Cheetah-2.4.4/cheetah/CheetahWrapper.py
x Cheetah-2.4.4/cheetah/Compiler.py
x Cheetah-2.4.4/CHANGES
x Cheetah-2.4.4/MANIFEST.in
x Cheetah-2.4.4/SetupTools.py
x Cheetah-2.4.4/SetupConfig.py
x Cheetah-2.4.4/bin/
x Cheetah-2.4.4/bin/cheetah
x Cheetah-2.4.4/bin/cheetah-analyze
x Cheetah-2.4.4/bin/cheetah-compile
x Cheetah-2.4.4/README.markdown
x Cheetah-2.4.4/LICENSE
x Cheetah-2.4.4/PKG-INFO
x Cheetah-2.4.4/setup.cfg
tar: Error exit delayed from previous errors.`

https://gist.github.com/ab7dea4d20ac6608584658955b8a3318

There's not much in the gist logs.  However if you look at the bolded lines looks like it tries to link two files with a similar name.  Then tar has a fit and dies.

**Comment by agozie** [comment]
`x Cheetah-2.4.4/cheetah/c/cheetah.h: Can't unlink already-existing object`
I have this same issue upgrading MongoDB.

**Comment by mikestecker** [comment]
I'm having the same issue as @agozie

**Comment by samcarecho** [comment]
Same issue here.

**Comment by SMillerDev** [comment]
Can someone please give the required information as described in the troubleshooting guide linked above? Saying you have the same issue without giving the information required to fix it won't help anyone.

**Comment by Andyparr** [comment]
- [X] - Run brew update twice.
- [X] - Run brew doctor and fix all the warnings (outdated Xcode/CLT and unbrewed dylibs are very likely to cause problems).
- [X] - Check that Command Line Tools for Xcode (CLT) and Xcode are up to date.
- [X] - If commands fail with permissions errors, check the permissions of /usr/local’s subdirectories. If you’re unsure what to do, you can run cd /usr/local && sudo chown -R $(whoami) bin etc include lib sbin share var Frameworks.
- [X] - Read through the Common Issues.

https://gist.github.com/329f0fc37b0529043d6c7f5500306d46

### **BREW CONFIG**
```
HOMEBREW_VERSION: 1.6.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0aab25bdb3bfa96d8eb14a38001ff6a24587bcdd
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bde1292d5f98949e3f566b78c05e08bd30914afe
Core tap last commit: 33 minutes ago
HOMEBREW_PREFIX: /usr/local
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.0 => /Applications/Xcode-beta.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 10, 1.8.0_91
macOS: 10.14-x86_64
CLT: 10.0.0.0.1.1530455945
CLT headers: 10.0.0.0.1.1530455945
Xcode: 10.0 => /Applications/Xcode-beta.app/Contents/Developer
XQuartz: N/A
```
### **BREW DOCTOR**
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /Applications/Postgres.app/Contents/Versions/latest/bin/gdal-config

Warning: You are using macOS 10.14.
We do not provide support for this pre-release version.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.

Warning: Ruby version 2.3.7 is unsupported on 10.14. Homebrew
is developed and tested on Ruby 2.3.3, and may not work correctly
on other Rubies. Patches are accepted as long as they don't cause breakage
on supported Rubies.
```

**Comment by SMillerDev** [comment]
Thanks for filling It out @Andyparr

Is anyone having these issues with a non-beta macOS version? Cause as brew doctor mentions, we don't really support pretty release softwaren since it can change quite a lot between versions.

**Comment by mistydemeo** [comment]
> Cheetah-2.4.4/cheetah/c/cheetah.h: Can't unlink already-existing object

This is a known OS bug in all of the 10.14 betas so far. It will probably be fixed in a later beta.

**Comment by Andyparr** [comment]
Unsure if this will help anyone but forcing mongodb to pour from the bottle using the --force-bottle flag allows it to install as a temporary workaround

**Comment by mistydemeo** [comment]
@alfiepoleon We're aware of the issue and we don't need any more reports on it. It's an OS bug in the current beta, and Homebrew just needs to wait for it to be fixed at this point.

**Comment by lonkly** [comment]
It was a but in Mojave 1,2,3 and now 4. For me, solution` brew install mongodb --force-bottle ` worked fine.

**Comment by PureWhiteWu** [comment]
Same error here

**Comment by PureWhiteWu** [comment]
Same error here in 10.13.6 (17G65)

**Comment by mistydemeo** [comment]
This should be fixed as of 10.14 beta 7.

@PureWhiteWu Could you open a new issue with the full debug output? https://docs.brew.sh/Troubleshooting I think you may have a different issue.

**Comment by PureWhiteWu** [comment]
@mistydemeo okay, I will file an issue later.
